cypress/e2e/swap.test.ts

```diff
@@ -75,35 +77,88 @@ describe('Swap', () => {
+          .then((initialBalance) => {
+            cy.get('#swap-currency-output .open-currency-select-button').click()
+            cy.contains('USDC').click()
+            cy.get('#swap-currency-output .token-amount-input').clear().type(BALANCE_INCREMENT.toString())
+            cy.get('#swap-currency-input .token-amount-input').should('not.equal', '')
+            cy.get('#swap-button').click()
+            cy.get('#confirm-swap-or-send').click()
+            cy.get(getTestSelector('dismiss-tx-confirmation')).click()
+
+            // Check that the new balance is reflected in the UI
             cy.get('#swap-currency-output [data-testid="balance-text"]').should(
               'have.text',
               `Balance: ${initialBalance + BALANCE_INCREMENT}`
             )
 
-            // chain state check
+            // Check that the balance is incremented on chain
             cy.then(() => hardhat.getBalance(hardhat.wallet.address, USDC_MAINNET))
               .then((balance) => Number(balance.toFixed(1)))
               .should('eq', initialBalance + BALANCE_INCREMENT)
           })
-        })
-    })
+      })
+  })
+
+  it('should render an error when a transaction fails due to a passed deadline', () => {
+    const BALANCE_INCREMENT = 1
+    const DEADLINE_MINUTES = 30
+    cy.visit('/swap', { ethereum: 'hardhat' })
+      .hardhat()
+      .then((hardhat) => {
+        hardhat
+          .reset()
+          .then(() => hardhat.getBalance(hardhat.wallet.address, USDC_MAINNET))
+          .then((balance) => Number(balance.toFixed(1)))
+          .then((initialBalance) => {
+            // input swap info
+            cy.get('#swap-currency-output .open-currency-select-button').click()
+            cy.contains('USDC').click()
+            cy.get('#swap-currency-output .token-amount-input').clear().type(BALANCE_INCREMENT.toString())
+            cy.get('#swap-currency-input .token-amount-input').should('not.equal', '')
+
+            // set deadline to minimum (1 minute)
+            cy.get(getTestSelector('open-settings-dialog-button')).click()
+            cy.get(getTestSelector('deadline-input')).clear().type(DEADLINE_MINUTES.toString())
+            cy.get('body').click('topRight')
+            cy.get(getTestSelector('deadline-input')).should('not.exist')
+
+            cy.get('#swap-button').click()
+            cy.get('#confirm-swap-or-send').click()
+
+            // Mine a block past the deadline
+            cy.then(() => hardhat.provider.send('eth_getBlockByNumber', ['latest']))
+              .then((block) => block.timestamp * 2)
+              .then((timestampIncrease) => {
+                hardhat.provider.send('evm_increaseTime', [hexValue(timestampIncrease)])
+                hardhat.provider.send('hardhat_mine', [hexValue(1), hexValue(timestampIncrease)])
+              })
+
+            cy.then(() => hardhat.provider.send('hardhat_mine', ['0x1', '0xc'])).then(() => {
```

Comment:
I think this can all be replaced by `hardhat.mine(1, DEADLINE_MINUTES * 60 + 1)`.

Reply:
You may need to update `cypress-hardhat` for this method to be available.